source/MaterialXRuntime/Private/PvtApi.cpp

```diff
@@ -4,81 +4,183 @@
 //
 
 #include <MaterialXRuntime/Private/PvtApi.h>
+#include <MaterialXRuntime/Private/PvtObject.h>
 
 #include <MaterialXRuntime/RtApi.h>
 #include <MaterialXRuntime/RtSchema.h>
 #include <MaterialXRuntime/RtNodeDef.h>
+#include <MaterialXRuntime/RtNodeGraph.h>
+#include <MaterialXRuntime/RtNodeImpl.h>
+#include <MaterialXRuntime/RtTargetDef.h>
 #include <MaterialXRuntime/RtFileIo.h>
 
 namespace MaterialX
 {
 
 namespace
 {
-    static const RtToken libRootName("api_lib_root");
+    const RtToken DEFAULT_LIBRARY_NAME("default");
+    const string NUMBERS("0123456789");
 }
 
 void PvtApi::reset()
 {
     _createFunctions.clear();
-    _stages.clear();
-    _nodedefs.clear();
-    _nodeimpls.clear();
-
-    _libraryRootStage.reset();
     _libraries.clear();
-    _libraryRootStage = RtStage::createNew(libRootName);
+    _librariesOrder.clear();
+    _stages.clear();
+    _stagesOrder.clear();
 
     _unitDefinitions = UnitConverterRegistry::create();
 }
 
-void PvtApi::createLibrary(const RtToken& name)
+RtStagePtr PvtApi::loadLibrary(const RtToken& name, const FilePath& path, const RtReadOptions* options, bool forceReload)
 {
-    // If already loaded unload the old first,
-    // to support reloading of updated libraries.
-    if (getLibrary(name))
+    auto it = _libraries.find(name);
+    if (it != _libraries.end())
+    {
+        if (forceReload)
+        {
+            unloadLibrary(name);
+        }
+        else
+        {
+            throw ExceptionRuntimeError("A library named '" + name.str() + "' has already been loaded");
+        }
+    }
+
+    RtStagePtr stage = PvtStage::createNew(name);
```

Comment:
@feldstj : As we discussed we'll need to change this to allow library loading w/o creating a new stage. Guess if it does not exist then we fail, and stage creation is required outside of here.

Reply:
Out of curiosity, why do you need to reload without recreating the stage? Note that the old stage is unloaded above before a new is created. The reason I unload first is that we want to allow reloading of the same library without creating duplicates. And we want to support reloading after a library has been edited, in which case we want to clear out the old before we load the new.

Reply:
This is not the use case of reload but adding additional libraries without recreating the stage.
This is just a not for us to follow-up on.